### PR TITLE
chore: enable more Go linters

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

Enables and sorts remaining linters left to be enabled.

Please review the EE changes commit here https://github.com/determined-ai/determined-ee/pull/1118

## Test Plan

CI passes

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/.golangci.yml [hunk 1]
```diff
@@ -84,79 +84,66 @@ linters-settings:
 linters:
```
Comment (hamidzr) [inline master/.golangci.yml:84]: do you have a resolved list of which ones are effectively moving out of the disabled list through this PR?

**Review by hamidzr — APPROVED** [review]
looks good thanks for organizing these

File: master/.golangci.yml (hunk 1 above)
Comment (NicholasBlaskey) [reply master/.golangci.yml:84]: Yep should be 
```
        bidichk
        containedctx
        decorder
        durationcheck
        errname
        exhaustruct
        gofumpt
        gosimple
        grouper
        importas
        makezero
        nilerr
        noctx
        tenv
        tparallel
        usestdlibvars
```

**Comment by NicholasBlaskey** [comment]
Admin merging over failing test_launch_layer_mnist

**Pull request merged**